### PR TITLE
coturn: Support secrets file for configuring the static-auth-secret

### PR DESCRIPTION
**Pull request by MatthewCroughan**
###### Motivation for this change

This PR simply rebases https://github.com/NixOS/nixpkgs/pull/75263 on master.

> This change allows for the coturn shared secret to be stored in a file outside the nix store.
It also makes it easier to publish configurations in public VCS with fewer issues.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/coturn.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import ./make-test-python.nix ({ ... }: {
+  name = "coturn";
```
Comment (Mic92) [inline nixos/tests/coturn.nix:2]: Can you add `passthru.tests.coturn = nixosTests.coturn;` to the coturn package?

Comment (Mic92) [reply nixos/tests/coturn.nix:2]: Also this test should be linked in `nixos/tests/all-tests.nix`.

**Comment by MatthewCroughan** [comment]
I have tested this as working on 21.05 on a Hetzner server I own.

File: nixos/tests/coturn.nix (hunk 1 above)
Comment (MatthewCroughan) [reply nixos/tests/coturn.nix:2]: Is 6be0a41 correct?

File: nixos/modules/services/networking/coturn.nix [hunk 2]
```diff
@@ -293,42 +301,65 @@ in {
...
+            ) "cap_net_bind_service";
+          Restart = "on-abort";
+        };
+      };
+    systemd.tmpfiles.rules = [
+      "d  /run/coturn 0700 turnserver turnserver - -"
+    ];
+  }]));
+
+  passthru.tests.coturn = nixosTests.coturn;
```
Comment (Mic92) [inline nixos/modules/services/networking/coturn.nix:364]: This is the module. I meant the coturn package.

**Comment by Mic92** [comment]
@GrahamcOfBorg test coturn

File: nixos/modules/services/networking/coturn.nix (hunk 2 above)
Comment (MatthewCroughan) [reply nixos/modules/services/networking/coturn.nix]: I see. Thank you.

**Comment by MatthewCroughan** [comment]
I've adjusted the authorship and split the commits out appropriately.

**Pull request merged**